### DISCUSSION TITLE
What is a "link card," anyway?

### DISCUSSION THREAD

**Discussion by bgawalt** [MarshalX/atproto#65] [Questions] [answered]
I'm having fun porting my "what's Brian reading these days" bot from Mastodon/Twitter to Bluesky:

https://bsky.app/profile/brian.gawalt.com/post/3jxrxemjco62r

but my latest snag is, how should I handle that link to the spreadsheet that's the bot's source of truth on what I'm reading.

One clear option is to make use of rich text features to get the URL to actually render as a hyper link, as in this example:
https://github.com/MarshalX/atproto/blob/main/examples/advanced_usage/send_rich_text.py

But what I *really* want is what the BlueSky clients call a "link card."

In terms of the jargon used in this library, what is a "link card" called?

(I think, equivalently: when a BlueSky posts a post with a link card, how does ATProto implement that card? I'm assuming the card is represented as a protocol object, rather than just a bsky.app-exclusive render job.)

UPDATE (10:51PM California time): I'm looking at the schema in `app.bsky.feed.post` and -- maybe link cards really aren't a protocol-level object? https://atproto.com/lexicons/app-bsky-feed#appbskyfeedpost

**Comment by bgawalt** [comment, marked as answer]
OK, the actual answer is, "a link card is a `app.bsky.embed.external`":
https://atproto.com/lexicons/app-bsky-embed#appbskyembedexternal

Here's a JavaScript example of someone posting a post with a link card:
https://github.com/aliceisjustplaying/atproto-starter-kit/blob/main/index.ts

which I found via the discussion on this pull request to `atprototools`:
https://github.com/ianklatzco/atprototools/pull/4

**Reply by MarshalX** [reply]
Python example: https://github.com/MarshalX/atproto/blob/main/examples/advanced_usage/send_embed.py
